store/data_store.py

```diff
@@ -8,18 +11,22 @@ class DataStore():
   def __init__(self,
       dataset_dir,
       transforms=[],
-      download=False
+      download=False,
+      max_samples=1,
```

Comment:
You've ignored the two level memory hierarchy. max_samples (and sample_size) at what level?